src/modules/metrics/MetricsConstants.ts

```diff
@@ -0,0 +1,47 @@
+export const METRICS_NAMES = {
+    // Counts
+    BROKER_COUNT: 'aggregator_broker_count',
+    CACHE_COUNT: 'aggregator_cache_count',
+    ERRORS_COUNT: 'aggregator_errors_count',
+    EXTERNAL_REQUESTS_COUNT: 'aggregator_external_requests_count',
+    INSTANCES_COUNT: 'aggregator_instances_count',
+    QUEUE_MESSAGES_COUNT: 'aggregator_queue_messages_count',
+    ZONE_MESSAGE_COUNT: 'aggregator_zone_message_count',
+
+    // Gauges
+    CACHE_GAUGE: 'aggregator_cache_gauge',
```

Comment:
The postfix should say something about the variable. Gauge as a postfix seems inaccurate. The whole variable btw should be self describing encapsulating all properties that the variable represents. This one for example should be `cache_current_items_count`.

Below you also have the postfix `_histogram` which is even more inaccurate as histograms add their own postfixes. I know it is only semantics, but for Prometheus it is important.

Reply:
691f5257777e15bc7ecceca2ebeedf2f10a87035 and e9aa20201e3ecf99abc8e6b5966aac1ef8a56c62 to better describe some odd ones

> Below you also have the postfix _histogram which is even more inaccurate as histograms add their own postfixes. I know it is only semantics, but for Prometheus it is important.

Not sure if I'm misunderstanding but it doesn't become `aggregator_external_requests_histogram_bucket_histogram`, it's `aggregator_external_requests_histogram_bucket` which is correct?

Reply:
The misconception starts because a counter usually ends with `_count`. The naming should be like `<namspace>_<description>_<metric_description>`. Metric description should be related to what it is sum, count, bytes, total, etc. For a histogram and also summary prometheus adds its own `_bucket`, `_sum`, `_count`. The generated prometheus description of the metric group will include that it is of the type histogram.

Example from default node metrics:
```
# HELP nodejs_gc_duration_seconds Garbage collection duration by kind, one of major, minor, incremental or weakcb.
# TYPE nodejs_gc_duration_seconds histogram
nodejs_gc_duration_seconds_bucket{le="0.001",kind="minor"} 287
nodejs_gc_duration_seconds_bucket{le="0.01",kind="minor"} 388
nodejs_gc_duration_seconds_bucket{le="0.1",kind="minor"} 388
nodejs_gc_duration_seconds_bucket{le="1",kind="minor"} 388
nodejs_gc_duration_seconds_bucket{le="2",kind="minor"} 388
nodejs_gc_duration_seconds_bucket{le="5",kind="minor"} 388
nodejs_gc_duration_seconds_bucket{le="+Inf",kind="minor"} 388
nodejs_gc_duration_seconds_sum{kind="minor"} 0.38074574875831596
nodejs_gc_duration_seconds_count{kind="minor"} 388
nodejs_gc_duration_seconds_bucket{le="0.001",kind="incremental"} 4
nodejs_gc_duration_seconds_bucket{le="0.01",kind="incremental"} 4
nodejs_gc_duration_seconds_bucket{le="0.1",kind="incremental"} 4
nodejs_gc_duration_seconds_bucket{le="1",kind="incremental"} 4
nodejs_gc_duration_seconds_bucket{le="2",kind="incremental"} 4
nodejs_gc_duration_seconds_bucket{le="5",kind="incremental"} 4
nodejs_gc_duration_seconds_bucket{le="+Inf",kind="incremental"} 4
nodejs_gc_duration_seconds_sum{kind="incremental"} 0.000738584041595459
nodejs_gc_duration_seconds_count{kind="incremental"} 4
nodejs_gc_duration_seconds_bucket{le="0.001",kind="major"} 0
nodejs_gc_duration_seconds_bucket{le="0.01",kind="major"} 2
nodejs_gc_duration_seconds_bucket{le="0.1",kind="major"} 2
nodejs_gc_duration_seconds_bucket{le="1",kind="major"} 2
nodejs_gc_duration_seconds_bucket{le="2",kind="major"} 2
nodejs_gc_duration_seconds_bucket{le="5",kind="major"} 2
nodejs_gc_duration_seconds_bucket{le="+Inf",kind="major"} 2
nodejs_gc_duration_seconds_sum{kind="major"} 0.005652751088142395
nodejs_gc_duration_seconds_count{kind="major"} 2
```

You can always break with these conventions, but other people will have given this a great deal of though plus people will know what to expect with these conventions.